twister: error on invalid platforms

nashif opened the pull request:
In case of unknown platforms in filters, raise an exception.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
twister: error on invalid platforms

In case of unknown platforms in filters such as plaform_allow,
integration_plaforms or platform_exclude, report an error and exit.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

golowanow commented on scripts/pylib/twister/twisterlib/testplan.py:
```diff
@@ -1261,12 +1249,14 @@ def verify_platforms_existence(self, platform_names_to_verify, log_info=""):
+                _platforms.append(self.get_platform(platform).name)
             else:
                 logger.error(f"{log_info} - unrecognized platform - {platform}")
                 sys.exit(2)
```
> In case of unknown platforms in filters, raise an exception.

this commit summary is misleading: no exception, but just log and exit.

nashif replied:
yeah, it was an exception in the first try...

golowanow commented on scripts/pylib/twister/twisterlib/testplan.py:
```diff
@@ -1261,12 +1249,14 @@ def verify_platforms_existence(self, platform_names_to_verify, log_info=""):
         for platform in platform_names_to_verify:
             if platform in self.platform_names:
-                continue
+                _platforms.append(self.get_platform(platform).name)
```
when `get_platform()` don't find name in aliases, then the resulting list will have None(s) instead of raising an error.

golowanow replied:
.. unless `self.platform_names` has all platform aliases already, but why to call `get_platform()` then ?

nashif replied:
we need the full platform name, not the alias or shorthand

nashif committed:
twister: testplan: do not verify_platforms_existence again

We already sanitize those lists very early on, no need to call this
check again on the same lists.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

golowanow commented on scripts/pylib/twister/twisterlib/testplan.py:
```diff
@@ -1261,12 +1234,16 @@ def verify_platforms_existence(self, platform_names_to_verify, log_info=""):
             else:
                 logger.error(f"{log_info} - unrecognized platform - {platform}")
                 sys.exit(2)
+        return _platforms
```
with this additional behavior the result needs to be used at apply_filters(), from line 901:
```
                platforms_ = self.verify_platforms_existence(ts.platform_allow, f"{ts_name} - platform_allow")
                a = set(platform_scope)
                b = set(filter(lambda item: item.name in platforms_, self.platforms))
                c = a.intersection(b)
                if not c:
                    platform_scope = list(
                        filter(lambda item: item.name in platforms_, self.platforms)
                    )
```

nashif replied:
this call needs to be removed, we do this verification already, so this is
another duplicate, looks like I missed it.
